### PR TITLE
Permanent fix for the issue #257

### PR DESCRIPTION
**Pull request by hishamco**
The PR #264 was a temporary fix for the issue #257, I create this PR to be a permanent fix for that previous issue

/cc @ryanbrandenburg @pranavkm

### CODE REVIEW

**Comment by hishamco** [comment]
I had to add a new property `Count` to `IResourceNamesCache` to make enumerator caches tests passing, I know this is a **breaking change** but I don't find a way to make all the tests passed without adding this property. Let me know if there's another a better way for doing this

**Comment by hishamco** [comment]
@pranavkm could you please have a look to this too, i'm not sure if there's a breaking should go in 2.0.0 or we need to wait until 3.0.0

**Comment by pranavkm** [comment]
@rynowak?

File: src/Microsoft.Extensions.Localization/Internal/AssemblyResourceStringProvider.cs [hunk 1]
```diff
@@ -1,115 +0,0 @@
...
-using System.Collections;
-using System.Collections.Generic;
-using System.Globalization;
-using System.IO;
-using System.Reflection;
-using System.Resources;
-
-namespace Microsoft.Extensions.Localization.Internal
-{
-    public class AssemblyResourceStringProvider : IResourceStringProvider
```
Comment (hishamco) [inline src/Microsoft.Extensions.Localization/Internal/AssemblyResourceStringProvider.cs]: I didn't get you?!!

**Comment by natemcmaster** [comment]
As you may already know, we're going to archive this repo. I'm working on moving the code today. Once complete, the Microsoft.Extensions.\* projects will be in https://github.com/aspnet/Extensions, and Microsoft.AspNetCore.\* will be in https://github.com/aspnet/AspNetCore. If this is something you'd like to get merged, please open a new PR against the new repos.

Thanks,
Nate

**Pull request closed without merging**